### DISCUSSION TITLE
Support <alias> = <expression> syntax like in T-SQL

### DISCUSSION THREAD

**Discussion by gonsalu** [duckdb/duckdb#3647] [General]
In T-SQL, one can write a query using "left-handed aliasing":

```
SELECT alias = expression
```

... in addition to the standard SQL way:

```
SELECT expression AS alias
```

When expressions are really long, I find it way more readable to be able to scan the aliases on the left side, instead of searching for the alias in the right hand side, which is typically not aligned.

Would it be possible to support this kind of syntax as well?

**Comment by Mytherin** [comment]
How do you resolve the ambiguity that is created when doing something like:

```sql
SELECT a=b
FROM tbl;
```

Is this "column b aliased to a" or "a equal to b"?

**Comment by gonsalu** [comment]
In SQL Server (T-SQL), it is "column b aliased to a":

![image](https://user-images.githubusercontent.com/126950/168310880-ffa31d60-f631-46af-b907-ca8eed36eb87.png)

**Reply by Mytherin** [reply]
That ambiguity makes it quite ugly to add this syntax in my opinion. We can't have the exact same behavior as SQL server there, since that would break query compatibility with all other SQL systems that would resolve this query as a comparison in the `SELECT` clause.

When there is no ambiguity we could perform the alias (i.e. when doing something like `SELECT c=b FROM tbl` where `tbl` has the columns `a` and `b`) but this sounds like a bad idea as well - adding a new table to the `FROM` clause can then unexpectedly change the syntax in the `SELECT` clause from an alias to a comparison, and aliasing columns to itself will not work unexpectedly as well (e.g. `a=a+1` will become a comparison).

**Reply by gonsalu** [reply]
Understandable.

Thought I'd give it a shot, since I use it a lot, and miss it in other SQL flavors.

**Comment by danshryock** [comment]
When I first saw DuckDB and all of the amazing additions it provides to the SQL language, this was one of the first things I looked for.  Changing the order of the alias is one of the best readability enhancements to SQL I've seen.  

While browsing through the syntax for macros I noticed that parameters can have default values:
`CREATE MACRO add_default(a, b := 5) AS a + b;`.

It seems like using `:=` as the operator between the name and the expression would be a good solution to resolve the ambiguity, and it would be consistent with other parts of DuckDB as well.
